notation.go

```diff
@@ -152,6 +165,44 @@ func Sign(ctx context.Context, signer Signer, repo registry.Repository, signOpts
 	return targetDesc, nil
 }
 
+func SignBlob(ctx context.Context, signer BlobSigner, reader io.Reader, signOpts SignBlobOptions) ([]byte, error) {
+	// sanity checks
+	if err := validate(signer, signOpts.SignerSignOptions); err != nil {
+		return nil, err
+	}
+	if reader == nil {
+		return nil, errors.New("reader cannot be nil")
+	}
+	if signOpts.ContentMediaType != "" {
+		if _, _, err := mime.ParseMediaType(signOpts.ContentMediaType); err != nil {
+			return nil, fmt.Errorf("invalid content media-type '%s': %v", signOpts.ContentMediaType, err)
+		}
+	}
+
+	sig, _, err := signer.SignBlob(ctx, reader, signOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return sig, nil
+}
+
+func validate(signer any, signOpts SignerSignOptions) error {
+	if signer == nil {
+		return errors.New("signer cannot be nil")
+	}
+	if signOpts.ExpiryDuration < 0 {
+		return fmt.Errorf("expiry duration cannot be a negative value")
+	}
+	if signOpts.ExpiryDuration%time.Second != 0 {
+		return fmt.Errorf("expiry duration supports minimum granularity of seconds")
+	}
+	if _, _, err := mime.ParseMediaType(signOpts.SignatureMediaType); err != nil {
```

Comment:
Do we need to validate the `SignatureMediaType` is equal to `jws.MediaTypeEnvelope` or `cose.MediaTypeEnvelope` directly?